File: B2C2C_Back_end/dockerized.postgresql/src/main/java/com/TFG_backend/dockerized/postgresql/aplicacion/AplicacionOfertaRepository.java
package com.TFG_backend.dockerized.postgresql.aplicacion;

import org.springframework.data.jpa.repository.JpaRepository;
import java.util.List;

public interface AplicacionOfertaRepository extends JpaRepository<AplicacionOferta, Long> {
    List<AplicacionOferta> findByAlumnoId(Long alumnoId);
    List<AplicacionOferta> findByOfertaId(Long ofertaId);
}

File: B2C2C_Front_end/app/src/main/java/com/torre/b2c2c_tfg/domain/repository/AplicacionOfertaRepository.kt
package com.torre.b2c2c_tfg.domain.repository

import com.torre.b2c2c_tfg.data.model.AplicacionOferta

interface AplicacionOfertaRepository {
    suspend fun crearAplicacion(aplicacion: AplicacionOferta): Boolean
    suspend fun getAplicacionesPorAlumnoId(alumnoId: Long): List<AplicacionOferta>
}